rocksdb_replicator/replicated_db.cpp

```diff
@@ -215,15 +266,18 @@ void RocksDBReplicator::ReplicatedDB::pullFromUpstream() {
           auto eb = db->client_->getChannel()->getEventBase();
           // It is very bad if we fail to rescheudle a pull request, we'd prefer
           // crashing.
-          eb->runInEventBaseThread([eb, weak_db = std::move(weak_db)] {
-              eb->runAfterDelay([weak_db = std::move(weak_db)] {
+          auto delay = FLAGS_replicator_pull_delay_on_error_ms;
+          // Randomize the delay so that helix (zk) is not overloaded from ext view requests.
+          auto randomized_delay = folly::Random::rand32(delay, delay * 2);
+          eb->runInEventBaseThread([eb, weak_db = std::move(weak_db), randomized_delay] {
+              eb->runAfterDelay([weak_db = std::move(weak_db), randomized_delay] {
```

Comment:
this lambda doesn't need to capture randomized_delay?

how about we just compute randomized_delay in the lambda and avoid capturing it in any of the lambdas